callstats-java-sdk/src/main/java/io/callstats/sdk/CallStats.java

```diff
@@ -322,7 +333,63 @@ public void onFailure(Exception e) {
 	}
 	
 	
-	public synchronized void sendCallStatsConferenceStats(String stats, UserInfo userInfo) {
+	public synchronized void startStatsReportingForUser(String userID, String confID) {
+		if (userID == null || confID == null) {
+			logger.error("startStatsReportingForUser: Arguments cannot be null ");
+			throw new IllegalArgumentException("startStatsReportingForUser: Arguments cannot be null");
+		}
+		
+		String key = userID+":"+confID;
+		
+		List<ConferenceStats> tempStats = conferenceStatsMap.get(userID);
+		if (tempStats == null) {
+			tempStats = new ArrayList<ConferenceStats>();
+			conferenceStatsMap.put(key, tempStats);
+		}
+	}
+
+	public synchronized void stopStatsReportingForUser(String userID, String confID) {
+		if (userID == null || confID == null) {
+			logger.error("stopStatsReportingForUser: Arguments cannot be null ");
+			throw new IllegalArgumentException("stopStatsReportingForUser: Arguments cannot be null");
+		}
+		String key = userID+":"+confID;
+		List<ConferenceStats> tempStats = conferenceStatsMap.get(key);
+		
+		if (tempStats != null && tempStats.size() > 0) {
+			StreamStats streamStats;
+			StreamStatsData streamStatsData;
+			ConferenceStats conferenceStats = tempStats.get(0);
+			ConferenceStatsData conferenceStatsData = new ConferenceStatsData(conferenceStats.getLocalUserID());
+			UserInfo info = new UserInfo(conferenceStats.getConfID(), conferenceStats.getLocalUserID(), conferenceStats.getUcID());
+			for (ConferenceStats stats : tempStats) {
+				streamStatsData = new StreamStatsData(stats.getRtt(), stats.getPacketsSent(), stats.getBytesSent(), stats.getJitter());
+				streamStats = new StreamStats(stats.getFromUserID(), stats.getToUserID(), stats.getStatsType(), streamStatsData);
+				conferenceStatsData.addStreamStats(stats.getSsrc(), streamStats);
+			}
+
+			String statsString = gson.toJson(conferenceStatsData);
+			logger.debug("Stats string is " + statsString);
+			sendCallStatsConferenceStats(statsString, info);
+		}
+	}
+
+	public synchronized void sendConferenceStats(String userID, ConferenceStats stats) {
+		if (stats == null || userID == null) {
+			logger.error("sendConferenceStats: Arguments cannot be null ");
+			throw new IllegalArgumentException("sendConferenceStats: Arguments cannot be null");
+		}
+		String key = userID+":"+stats.getConfID();
+		List<ConferenceStats> tempStats = conferenceStatsMap.get(key);
+		if (tempStats == null) {
```

Comment:
If this is null, "startStatsReportingForUser" wasn't called. I think we should throw an exception here